core/app/services/uffizzi_core/activity_item_service.rb

```diff
@@ -27,8 +27,9 @@ def disable_deployment!(activity_item)
 
     def fail_deployment!(activity_item)
       deployment = activity_item.container.deployment
+      last_event = activity_item.events.order_by_id.last
 
-      activity_item.events.create(state: UffizziCore::Event.state.failed)
+      activity_item.events.create(state: UffizziCore::Event.state.failed) if last_event&.state != UffizziCore::Event.state.failed
```

Comment:
Will there be cases when this condition is met? If it was called from the open-source app?

Reply:
Yes, actually, I add this after a platform [check](https://github.com/UffizziCloud/uffizzi_platform/actions/runs/3238410745/jobs/5306674039) fails. The [platform activity items service](https://github.com/UffizziCloud/uffizzi_platform/blob/a5e4dd7e60d09f840eed6eab920f93cb9b625968/app/services/activity_item_service.rb#L28) creates an event with failed state and when `UffizziCore::ActivityItemService.fail_deployment!(activity_item)` is invoked tries to create another, so I added this validation

Reply:
Can we do like this?
`activity_item.events.create(state: UffizziCore::Event.state.failed) unless last_event&.failed`